frontend/result/script.js

```diff
@@ -4,19 +4,55 @@ let returnButton = document.querySelector(".main__return-button");
 let city = params.get("city");
 let url = document.location.origin + `/api/get/weather/${city}`;
 let resPage = document.querySelector(".res-page")
+
 returnButton.addEventListener("click", () => {
   window.location.replace(window.location.origin);
 });
+
+class Weather {
+  constructor(temperature, description, feels_like, country, humidity, wind_speed) {
+    this.temperature = temperature
+    this.descriprtion = description
+    this.feels_like = feels_like
+    this.country = country
+    this.humidity = humidity
+    this.wind_speed = wind_speed
+  }
+  match() {
+    switch (this.descriprtion) {
+      case 'Rain':
+        return "/assets/icons/free-icon-storm-578132.png"
+      case 'Clouds':
+        return "/assets/icons/free-icon-cloud-4814293.png"
+      case 'Thunderstorm':
+        return "/assets/icons/free-icon-storm-3026395.png"
+      case 'Clear':
+        return "/assets/icons/free-icon-sun-4814268.png"
+      case 'Mist':
+        return "/assets/icons/free-icon-fog-3750506.png"
+      case 'Haze':
+        return "/assets/icons/free-icon-fog-3750506.png"
+    }
+  }
+  generImg() {
+    if (this.country !== "UA") {
+      return `https://flagsapi.com/${this.country}/flat/64.png`
+    }
+    return `https://flagsapi.com/RU/flat/64.png`
```

Comment:
pls refactor this(more logical then now)

now:
```
if not some_specific_situation:
    return all
return some_specific_result
```
will be
```
if some_specific_situation:
    return some_specific_result
return all
```

Reply:
done